src/request.rs

```diff
@@ -200,75 +202,56 @@ impl<B> RequestBuilder<B> {
         self.header(http::header::AUTHORIZATION, format!("Bearer {}", token.into()))
     }
 
-    fn body(self, body: impl AsRef<[u8]>) -> RequestBuilder<impl AsRef<[u8]>> {
-        RequestBuilder {
-            url: self.url,
-            method: self.method,
-            headers: self.headers,
-            body,
-            max_redirections: self.max_redirections,
-            follow_redirects: self.follow_redirects,
-            #[cfg(feature = "charsets")]
-            default_charset: self.default_charset,
-            #[cfg(feature = "compress")]
-            allow_compression: self.allow_compression,
-        }
-    }
-
     /// Set the body of this request to be text.
     ///
     /// If the `Content-Type` header is unset, it will be set to `text/plain` and the carset to UTF-8.
-    pub fn text(mut self, body: impl AsRef<str>) -> RequestBuilder<impl AsRef<[u8]>> {
-        struct Text<B1>(B1);
-
-        impl<B1: AsRef<str>> AsRef<[u8]> for Text<B1> {
-            fn as_ref(&self) -> &[u8] {
-                self.0.as_ref().as_bytes()
-            }
-        }
-
+    pub fn text(mut self, body: &'body str) -> Self {
         self.headers
             .entry(http::header::CONTENT_TYPE)
             .unwrap()
             .or_insert(HeaderValue::from_static("text/plain; charset=utf-8"));
-        self.body(Text(body))
+        self.body = Cow::Borrowed(body.as_bytes());
+        self
     }
 
     /// Set the body of this request to be bytes.
     ///
     /// If the `Content-Type` header is unset, it will be set to `application/octet-stream`.
-    pub fn bytes(mut self, body: impl AsRef<[u8]>) -> RequestBuilder<impl AsRef<[u8]>> {
+    pub fn bytes(mut self, body: &'body [u8]) -> Self {
         self.headers
             .entry(http::header::CONTENT_TYPE)
             .unwrap()
             .or_insert(HeaderValue::from_static("application/octet-stream"));
-        self.body(body)
+        self.body = Cow::Borrowed(body);
+        self
     }
 
     /// Set the body of this request to be the JSON representation of the given object.
     ///
     /// If the `Content-Type` header is unset, it will be set to `application/json` and the charset to UTF-8.
     #[cfg(feature = "json")]
-    pub fn json<T: serde::Serialize>(mut self, value: &T) -> Result<RequestBuilder<impl AsRef<[u8]>>> {
+    pub fn json<T: serde::Serialize>(mut self, value: &T) -> Result<Self> {
```

Comment:
Since this is an owned variant, shouldn't it erase the lifetime and return `RequestBuilder<'static>`?